### DISCUSSION TITLE
Dynamic path does not generate HTML even with getStaticProps

### DISCUSSION THREAD

**Discussion by angelod1as** [vercel/next.js#60115] [Help]
### Summary

TL;DR:

I have some pages being ISR instead of SSG even without `revalidate` in `getStaticProps`.

How to force them being SSG and not ISR on build time?

## Context

I have a `[...path].tsx` file that uses `getStaticPaths` and `getStaticProps`. When I run `yarn build` it even shows me the SSG `●` 

![image](https://github.com/vercel/next.js/assets/13950513/a5ee43a1-e1ea-40e9-b6c8-ce227d70701f)

I realise this might be ISR, but below, in the build section, this is the message shown:

```
●  (SSG)      prerendered as static HTML (uses getStaticProps)
   (ISR)      incremental static regeneration (uses revalidate in getStaticProps)
```

I do not have `revalidate` in my getStaticProps and/or paths:

```tsx
export const getStaticPaths: GetStaticPaths = async (context) => {
  const cmsPaths = await fetchCmsPaths()
  const locales = context.locales || []

  const paths = cmsPaths
    .filter((path) => path !== "/" && path !== "/404" && path !== "/500")
    .map((path) =>
      locales.map((locale) => ({
        params: { path: path.split("/").filter(Boolean) },
        locale,
      })),
    )
    .flat()

  return {
    paths,
    fallback: "blocking",
  }
}
```

```tsx
export const getStaticProps: GetStaticProps<ContentPageProps> = async (
  context,
) => {
  const segments = context.params?.path as string[]
  const path = segments.map((segment) => `/${segment}`).join("")

  const cmsPage = await fetchCmsPage({ path })

  if (!cmsPage) {
    return {
      notFound: true,
    }
  }

  return {
    props: {
      cmsPage,
    },
  }
}
```

When I visit the `.next/server/pages` folder, instead of having the pre-built HTML (like my `index.html` and others), I have just this:

![image](https://github.com/vercel/next.js/assets/13950513/1da75cbb-afd3-4253-86a7-ab147ca97b2f)

The moment I run `yarn start` and visit one of the pages caught by `[...path].tsx` file, a HTML magically appears inside the folder.

For very specific project reasons, I need *all* the HTML to be pre-built and not inscrementally generated like it's happening right now.

How can I achieve that?
